cyral/resource_cyral_repository_identity_map.go

```diff
@@ -218,37 +232,63 @@ func resourceRepositoryIdentityMap(deprecate bool) *schema.Resource {
 						d.Get("identity_name").(string),
 						d.Get("repository_local_account_id").(string))
 				},
-				ResourceData: &RepositoryIdentityMapResource{},
-				ResponseData: &RepositoryIdentityMapAPIResponse{},
+				ResourceData:   &RepositoryIdentityMapResource{},
+				ResponseData:   &RepositoryIdentityMapAPIResponse{},
+				UnsetIdOnError: true,
 			}, ReadRepositoryIdentityMapConfig,
 		),
 		ReadContext: ReadResource(ReadRepositoryIdentityMapConfig),
-		UpdateContext: UpdateResource(
-			ResourceOperationConfig{
-				Name:       "RepositoryIdentityMapResourceUpdate",
-				HttpMethod: http.MethodPut,
-				CreateURL: func(d *schema.ResourceData, c *client.Client) string {
-					return fmt.Sprintf("https://%s/v1/repos/%s/identityMaps/%s/%s/%s",
-						c.ControlPlane,
-						d.Get("repository_id").(string),
-						d.Get("identity_type").(string),
-						d.Get("identity_name").(string),
-						d.Get("repository_local_account_id").(string))
-				},
-				ResourceData: &RepositoryIdentityMapResource{},
-			}, ReadRepositoryIdentityMapConfig,
-		),
+		UpdateContext: func(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+			c := m.(*client.Client)
+			readConfig := ReadRepositoryIdentityMapConfig
+
+			repoIdNew := d.Get("repository_id")
+			idTypeNew := d.Get("identity_type")
+			idNameNew := d.Get("identity_name")
+			accIdNew := d.Get("repository_local_account_id")
+
+			resourceData := &RepositoryIdentityMapResource{}
+			resourceData.ReadFromSchema(d)
+
+			createNewUrl := fmt.Sprintf("https://%s/v1/repos/%s/identityMaps/%s/%s/%s",
+				c.ControlPlane,
+				repoIdNew,
+				idTypeNew,
+				idNameNew,
+				accIdNew,
+			)
+
+			if _, err := c.DoRequest(createNewUrl, http.MethodPut, resourceData); err != nil {
+				return createError("Unable to create new integration", fmt.Sprintf("%v", err))
+			}
+
+			d.SetId(fmt.Sprintf("%s/identityMaps/%s/%s/%s",
+				d.Get("repository_id").(string),
+				d.Get("identity_type").(string),
+				d.Get("identity_name").(string),
+				d.Get("repository_local_account_id").(string)),
+			)
```

Comment:
1. This will break backwards compatibility with previous resource. If this is the only way to fix the issue, then I'm ok with that.
2. So far, we used `-` as a separator for those IDs that are composed of several elements. We should use the same set of practices in the whole provider unless it is not possible;
3. Using the ID to compose the URL makes the code harder to read as you need to check how the ID is created in order to guess what is the resource endpoint.

```suggestion
			d.SetId(fmt.Sprintf("%s-%s-%s-%s",
				d.Get("repository_id").(string),
				d.Get("identity_type").(string),
				d.Get("identity_name").(string),
				d.Get("repository_local_account_id").(string)),
			)
```

Reply:
This just makes we need to have splits and joins all over the place. The `/`s are for the URL. This is a pattern among other providers such as gcloud and other things, and most of our IDs are actual IDs, and that's why we use `-`, but in this case it's different, and having `-` just makes things more complicated, in my eye.